app/controllers/authentication_controller.rb

```diff
@@ -50,10 +72,23 @@ def trigger_successful_support_user_sign_in_event(sign_in_type, oid = nil)
   end
 
   def trigger_failed_dsi_sign_in_event(sign_in_type, oid = nil)
+    event = DfE::Analytics::Event.new
+      .with_type(:failed_dsi_sign_in_attempt)
+      .with_request_details(request)
+      .with_response_details(response)
+      .with_user(current_user)
+      .with_data(user_anonymised_id: StringAnonymiser.new(oid), sign_in_type: sign_in_type)
+
+    DfE::Analytics::SendEvents.do([event])
+
     request_event.trigger(
       :failed_dsi_sign_in_attempt,
       user_anonymised_id: StringAnonymiser.new(oid),
```

Comment:
Not sure if field `user_anonymised_id` is required and please see comment about id, oid and DfE:Analytics.anonymiser() above.